### ISSUE TITLE
Missing Library when building Magma server

### ISSUE THREAD

**Issue by DaanSchar** [itzg/docker-minecraft-server#1894]
### Describe the problem

It seems that a library is being retreived from the following location: https://maven.creeperhost.net/io/izzel/tools/1.3.0/tools-1.3.0.jar. This returns a 404 status code and the container is not able to complete the init phase.

### Container definition

docker-compose.yml

```
version: '3'

services:
  minecraft:
    image: itzg/minecraft-server
    container_name: magma_server
    volumes:
      - ./minecraft-data:/data
      - ./mods:/mods
    ports:
      - 25565:25565
      - 24454:24454/udp
    environment:
      USE_AIKAR_FLAGS: "TRUE"
      EULA: "TRUE"
      SERVER_NAME: "Magma"
      TYPE: "MAGMA"
      VERSION: "1.18.2"
      MEMORY: "8G"
    tty: true
    stdin_open: true
    restart: unless-stopped

volumes:
  minecraft-data:
    external: true
```

### Container logs

```
lords-and-legions  | [init] Running as uid=1000 gid=1000 with /data as 'drwxrwxr-x 10 1000 1000 4096 Jan 11 21:03 /data'
lords-and-legions  | [init] Resolved version given 1.18.2 into 1.18.2 and major version 1.18
lords-and-legions  | [init] Resolving type given MAGMA
lords-and-legions  | [init] Installing Forge 1.18.2-bf1bd4bc. This might take a minute or two...
lords-and-legions  | [init] Install failed. Trying again...
lords-and-legions  | [init] Install failed. Trying again...
lords-and-legions  | JVM info: Eclipse Adoptium - 17.0.3 - 17.0.3+7
lords-and-legions  | java.net.preferIPv4Stack=true
lords-and-legions  | Found java version 17.0.3
lords-and-legions  | Target Directory: .
lords-and-legions  | Data kindly mirrored by CreeperHost at https://www.creeperhost.net/
lords-and-legions  | Considering minecraft server jar
lords-and-legions  | Downloading libraries
lords-and-legions  | Found 0 additional library directories
lords-and-legions  | Considering library cpw.mods:securejarhandler:1.0.5
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library org.ow2.asm:asm:9.4
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library org.ow2.asm:asm-commons:9.4
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library org.ow2.asm:asm-tree:9.4
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library org.ow2.asm:asm-util:9.4
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library org.ow2.asm:asm-analysis:9.4
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library net.minecraftforge:accesstransformers:8.0.4
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library org.antlr:antlr4-runtime:4.9.1
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library net.minecraftforge:eventbus:5.0.3
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library net.minecraftforge:forgespi:4.0.15-4.x
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library net.minecraftforge:coremods:5.0.1
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library cpw.mods:modlauncher:9.1.3
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library net.minecraftforge:unsafe:0.2.0
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library com.electronwill.night-config:core:3.6.4
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library com.electronwill.night-config:toml:3.6.4
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library org.apache.maven:maven-artifact:3.6.3
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library net.jodah:typetools:0.8.3
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library net.minecrell:terminalconsoleappender:1.2.0
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library org.jline:jline-reader:3.21.0
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library org.jline:jline-terminal:3.21.0
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library org.spongepowered:mixin:0.8.5
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library org.openjdk.nashorn:nashorn-core:15.3
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library net.minecraftforge:JarJar:0.3.0
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library net.md-5:bungeecord-chat:1.16-R0.4
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library org.yaml:snakeyaml:1.30
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library org.fusesource.jansi:jansi:1.18
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library jline:jline:2.12.1
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library com.googlecode.json-simple:json-simple:1.1.1
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library org.xerial:sqlite-jdbc:3.36.0.3
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library mysql:mysql-connector-java:8.0.27
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library org.apache.logging.log4j:log4j-iostreams:2.17.1
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library net.md-5:SpecialSource:1.10.0
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library com.google.guava:guava:31.0.1-jre
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library com.google.code.gson:gson:2.8.8
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library org.magma:magma-common:2.6
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library io.izzel:tools:1.3.0
lords-and-legions  |   Downloading library from https://maven.creeperhost.net/io/izzel/tools/1.3.0/tools-1.3.0.jar
lords-and-legions  | java.io.FileNotFoundException: https://maven.creeperhost.net/io/izzel/tools/1.3.0/tools-1.3.0.jar
lords-and-legions  | 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
lords-and-legions  | 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
lords-and-legions  | 	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
lords-and-legions  | 	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Unknown Source)
lords-and-legions  | 	at java.base/java.lang.reflect.Constructor.newInstance(Unknown Source)
lords-and-legions  | 	at java.base/sun.net.www.protocol.http.HttpURLConnection$10.run(Unknown Source)
lords-and-legions  | 	at java.base/sun.net.www.protocol.http.HttpURLConnection$10.run(Unknown Source)
lords-and-legions  | 	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
lords-and-legions  | 	at java.base/sun.net.www.protocol.http.HttpURLConnection.getChainedException(Unknown Source)
lords-and-legions  | 	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(Unknown Source)
lords-and-legions  | 	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(Unknown Source)
lords-and-legions  | 	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(Unknown Source)
lords-and-legions  | 	at net.minecraftforge.installer.DownloadUtils.download(DownloadUtils.java:186)
lords-and-legions  | 	at net.minecraftforge.installer.DownloadUtils.download(DownloadUtils.java:171)
lords-and-legions  | 	at net.minecraftforge.installer.DownloadUtils.downloadLibrary(DownloadUtils.java:159)
lords-and-legions  | 	at net.minecraftforge.installer.actions.Action.downloadLibraries(Action.java:89)
lords-and-legions  | 	at net.minecraftforge.installer.actions.ServerInstall.run(ServerInstall.java:114)
lords-and-legions  | 	at net.minecraftforge.installer.SimpleInstaller.main(SimpleInstaller.java:128)
lords-and-legions  | Caused by: java.io.FileNotFoundException: https://maven.creeperhost.net/io/izzel/tools/1.3.0/tools-1.3.0.jar
lords-and-legions  | 	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(Unknown Source)
lords-and-legions  | 	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(Unknown Source)
lords-and-legions  | 	at java.base/java.net.HttpURLConnection.getResponseCode(Unknown Source)
lords-and-legions  | 	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(Unknown Source)
lords-and-legions  | 	at net.minecraftforge.installer.DownloadUtils.getConnection(DownloadUtils.java:243)
lords-and-legions  | 	at net.minecraftforge.installer.DownloadUtils.download(DownloadUtils.java:184)
lords-and-legions  | 	... 5 more
lords-and-legions  |   Downloading library from https://repo1.maven.org/maven2/io/izzel/tools/1.3.0/tools-1.3.0.jar
lords-and-legions  | java.io.FileNotFoundException: https://repo1.maven.org/maven2/io/izzel/tools/1.3.0/tools-1.3.0.jar
lords-and-legions  | 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
lords-and-legions  | 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
lords-and-legions  | 	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
lords-and-legions  | 	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Unknown Source)
lords-and-legions  | 	at java.base/java.lang.reflect.Constructor.newInstance(Unknown Source)
lords-and-legions  | 	at java.base/sun.net.www.protocol.http.HttpURLConnection$10.run(Unknown Source)
lords-and-legions  | 	at java.base/sun.net.www.protocol.http.HttpURLConnection$10.run(Unknown Source)
lords-and-legions  | 	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
lords-and-legions  | 	at java.base/sun.net.www.protocol.http.HttpURLConnection.getChainedException(Unknown Source)
lords-and-legions  | 	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(Unknown Source)
lords-and-legions  | 	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(Unknown Source)
lords-and-legions  | 	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(Unknown Source)
lords-and-legions  | 	at net.minecraftforge.installer.DownloadUtils.download(DownloadUtils.java:186)
lords-and-legions  | 	at net.minecraftforge.installer.DownloadUtils.download(DownloadUtils.java:174)
lords-and-legions  | 	at net.minecraftforge.installer.DownloadUtils.downloadLibrary(DownloadUtils.java:159)
lords-and-legions  | 	at net.minecraftforge.installer.actions.Action.downloadLibraries(Action.java:89)
lords-and-legions  | 	at net.minecraftforge.installer.actions.ServerInstall.run(ServerInstall.java:114)
lords-and-legions  | 	at net.minecraftforge.installer.SimpleInstaller.main(SimpleInstaller.java:128)
lords-and-legions  | Caused by: java.io.FileNotFoundException: https://repo1.maven.org/maven2/io/izzel/tools/1.3.0/tools-1.3.0.jar
lords-and-legions  | 	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(Unknown Source)
lords-and-legions  | 	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(Unknown Source)
lords-and-legions  | 	at java.base/java.net.HttpURLConnection.getResponseCode(Unknown Source)
lords-and-legions  | 	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(Unknown Source)
lords-and-legions  | 	at net.minecraftforge.installer.DownloadUtils.getConnection(DownloadUtils.java:243)
lords-and-legions  | 	at net.minecraftforge.installer.DownloadUtils.download(DownloadUtils.java:184)
lords-and-legions  | 	... 5 more
lords-and-legions  | Considering library it.unimi.dsi:fastutil:8.5.6
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library cpw.mods:bootstraplauncher:1.0.0
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library net.minecraftforge:fmlloader:1.18.2-40.2.0
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library com.github.jponge:lzma-java:1.3
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library com.google.code.findbugs:jsr305:3.0.2
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library com.google.code.gson:gson:2.8.7
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library com.google.errorprone:error_prone_annotations:2.1.3
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library com.google.guava:guava:20.0
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library com.google.guava:guava:25.1-jre
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library com.google.j2objc:j2objc-annotations:1.1
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library com.nothome:javaxdelta:2.0.1
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library com.opencsv:opencsv:4.4
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library commons-beanutils:commons-beanutils:1.9.3
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library commons-collections:commons-collections:3.2.2
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library commons-io:commons-io:2.4
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library commons-logging:commons-logging:1.2
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library de.oceanlabs.mcp:mcp_config:1.18.2-20220404.173914@zip
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library de.siegmar:fastcsv:2.0.0
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library net.md-5:SpecialSource:1.11.0
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library net.minecraftforge:ForgeAutoRenamingTool:0.1.22:all
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library net.minecraftforge:binarypatcher:1.0.12
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library net.minecraftforge:fmlcore:1.18.2-40.2.0
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library net.minecraftforge:fmlloader:1.18.2-40.2.0
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library net.minecraftforge:forge:1.18.2-40.2.0:universal
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library net.minecraftforge:installertools:1.3.0
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library net.minecraftforge:jarsplitter:1.1.4
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library net.minecraftforge:javafmllanguage:1.18.2-40.2.0
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library net.minecraftforge:lowcodelanguage:1.18.2-40.2.0
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library net.minecraftforge:mclanguage:1.18.2-40.2.0
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library net.minecraftforge:srgutils:0.4.11
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library net.minecraftforge:srgutils:0.4.3
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library net.minecraftforge:srgutils:0.4.9
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library net.sf.jopt-simple:jopt-simple:5.0.4
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library net.sf.jopt-simple:jopt-simple:6.0-alpha-3
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library org.apache.commons:commons-collections4:4.2
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library org.apache.commons:commons-lang3:3.8.1
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library org.apache.commons:commons-text:1.3
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library org.checkerframework:checker-qual:2.0.0
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library org.codehaus.mojo:animal-sniffer-annotations:1.14
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library org.ow2.asm:asm-analysis:9.2
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions exited with code 1
lords-and-legions  | Considering library org.ow2.asm:asm-analysis:9.3
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library org.ow2.asm:asm-commons:9.2
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library org.ow2.asm:asm-commons:9.3
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library org.ow2.asm:asm-tree:9.2
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library org.ow2.asm:asm-tree:9.3
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library org.ow2.asm:asm:9.2
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library org.ow2.asm:asm:9.3
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | Considering library trove:trove:1.0.2
lords-and-legions  |   File exists: Checksum validated.
lords-and-legions  | These libraries failed to download. Try again.
lords-and-legions  | 
lords-and-legions  | io.izzel:tools:1.3.0
lords-and-legions  | There was an error during installation
lords-and-legions  | [init] 
lords-and-legions  | ERROR Forge failed to install after several tries.
lords-and-legions  | 

```

**Comment by itzg** [comment]
Please re-pull the latest image and try again. The logs you provided don't match the latest implementation, such as

```
> docker run -it --rm -e EULA=true -e TYPE=MAGMA -e VERSION=1.18.2 itzg/minecraft-server

[init] Running as uid=1000 gid=1000 with /data as 'drwxrwxr-x 2 1000 1000 4096 Dec 20 16:24 /data'
[init] Resolved version given 1.18.2 into 1.18.2 and major version 1.18
[init] Resolving type given MAGMA
[init] Creating server properties in /data/server.properties
[init] Disabling whitelist functionality
[init] Setting initial memory to 1G and max to 1G
[init] Starting the Minecraft server...
    __  ___                           
   /  |/  /___ _____ _____ ___  ____ _
  / /|_/ / __ `/ __ `/ __ `__ \/ __ `/
 / /  / / /_/ / /_/ / / / / / / /_/ / 
/_/  /_/\__,_/\__, /_/ /_/ /_/\__,_/  
             /____/                   
Copyright (c) 2023 MagmaFoundation.
--------------------------------------
Running on Java 17.0.5 (Eclipse Adoptium)
Magma version   1.18.2-bf1bd4bc
Bukkit version  v1_18_R2
Forge version   1.18.2-40.2.0-bf1bd4b
--------------------------------------
WARNING: The new server jar is still under development and will be unstable! If you experience any issues, please report them to the developers.
WARNING: If the server crashes while installing, try removing the libraries folder and launching the server again.
Loading libraries... 100% [========================] 79/79 (0:00:10 / 0:00:00) 
Patching server... 100% [============================] 8/8 (0:00:12 / 0:00:00) 
2023-01-11 22:32:46,930 main WARN Advanced terminal features are not available in this environment
```

**Comment by DaanSchar** [comment]
Updating the image seems to have done the trick.
